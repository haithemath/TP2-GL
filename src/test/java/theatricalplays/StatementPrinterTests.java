package theatricalplays;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new Play("Hamlet",PieceType.tragedy),
                "as-like", new Play("As You Like It", PieceType.comdey),
                "othello", new Play("Othello", PieceType.tragedy));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    }

    // dans notre cas, ce test ne sert à rien puisque on a déjà introduit un nouveau type énumerative qui
    // va protéger le type de théatre en acceptant que les valeurs definies par ce type-là

    // @Test
    // void statementWithNewPlayTypes() {
    //     Map<String, Play> plays = Map.of(
    //             "henry-v",  new Play("Henry V", PieceType.henry),
    //             "as-like", new Play("As You Like It", "pastoral"));

    //     Invoice invoice = new Invoice("BigCo", List.of(
    //             new Performance("henry-v", 53),
    //             new Performance("as-like", 55)));

    //     StatementPrinter statementPrinter = new StatementPrinter();
    //     Assertions.assertThrows(Error.class, () -> {
    //         statementPrinter.print(invoice, plays);
    //     });
    // }
}
