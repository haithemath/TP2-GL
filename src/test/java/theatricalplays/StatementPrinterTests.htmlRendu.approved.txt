<!DOCTYPE html>
<html>
<head>
<title>Invoice</title>
</head>
<body>
<h1>Invoice</h1>
<ul>
<li><strong>Client : </strong>BigCo</li>
</ul>
<table border=1 style= "border-spacing: 4px;">
<thead>
<tr>
<th>Piece</th>
<th>Seats sold</th>
<th>Price</th>
</tr>
</thead>
<tbody>
<tr>
<td>Hamlet</td>
<td>55</td>
<td>$650.00</td>
</tr>
<tr>
<td>As You Like It</td>
<td>35</td>
<td>$580.00</td>
</tr>
<tr>
<td>Othello</td>
<td>40</td>
<td>$500.00</td>
</tr>
<tr>
<th colspan="2">Total owed:</th>
<td>$1,730.00</td>
</tr>
<tr>
<th colspan="2">Fidelity points earned:</th>
<td>47</td>
</tr>
</tbody>
</table>
<p>Payement is required under 30 days. We can break your knees if you don't so.</p>
</body>
</html>
